google-oauth-client/src/main/java/com/google/api/client/auth/openidconnect/IdTokenVerifier.java

```diff
@@ -76,11 +123,17 @@ public IdTokenVerifier() {
 
   /** @param builder builder */
   protected IdTokenVerifier(Builder builder) {
+    this.certificatesLocation = builder.certificatesLocation;
     clock = builder.clock;
     acceptableTimeSkewSeconds = builder.acceptableTimeSkewSeconds;
     issuers = builder.issuers == null ? null : Collections.unmodifiableCollection(builder.issuers);
     audience =
         builder.audience == null ? null : Collections.unmodifiableCollection(builder.audience);
+    this.publicKeyCache =
+        CacheBuilder.newBuilder()
+            .expireAfterWrite(1, TimeUnit.HOURS)
+            .build(new PublicKeyLoader(builder.httpTransportFactory));
```

Comment:
`httpTransportFactory` is now required or you'll get a NPE later. We generally prefer to use preconditions to throw the NPE early for required attributes (using guava's `Preconditions.checkNotNull`)

Reply:
Good catch, thanks, I think here we want to set a default ourselves in case of null, otherwise it is going to be a breaking change?